src/raftstore/store/peer.rs

```diff
@@ -1241,14 +1237,115 @@ impl Peer {
         Ok(())
     }
 
-    fn handle_raft_committed_entries(&mut self,
-                                     committed_entries: Vec<eraftpb::Entry>)
-                                     -> Vec<ExecResult> {
+    fn exec_read(&mut self, req: &RaftCmdRequest) -> Result<RaftCmdResponse> {
+        let snap = Snapshot::new(self.engine.clone());
+        let requests = req.get_requests();
+        let mut responses = Vec::with_capacity(requests.len());
+
+        for req in requests {
+            let cmd_type = req.get_cmd_type();
+            let mut resp = match cmd_type {
+                CmdType::Get => try!(do_get(&self.tag, self.region(), &snap, req)),
+                CmdType::Snap => try!(do_snap(self.region().to_owned())),
+                CmdType::Put | CmdType::Delete | CmdType::Invalid => unreachable!(),
```

Comment:
No, as the macro says, it can't be reached. Because `should_read_local` has done the check already.